src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -42,6 +44,15 @@ def run(self):
                             log.info("thread [{0}] terminating due to reconfigure".format(thread_name))
                             self.async_job.threads.remove(self)
                             return
+
+                        if not self.async_job.run_event.is_set():
```

Comment:
This is not making any sense. Why have this check here and terminate the thread, but also wait() on the event in L#69?

Reply:
Changed.